homeassistant/components/streamlabswater/__init__.py

```diff
@@ -0,0 +1,112 @@
+"""
+Support for Streamlabs Water Monitor.
+
+This component provides sensors that return the daily, monthly, and yearly
+water usage as returned by the Streamlabs Water service. A binary sensor
+is used to indicate whether the water monitor is in the home or away mode.
+
+In addition to the sensors, a service is provided that can be used to set
+the away mode of the water monitor.
+
+The minimum configuration needed to get started is:
+
+    streamlabswater:
+      api_key: <your_api_key>
+
+where the api_key is retrieved by following the instructions at:
+https://developer.streamlabswater.com/docs/getting-started.html
+
+By default the first location found will be used. You can specify a location:
+
+    streamlabswater:
+      api_key: <your_api_key>
+      location_id: <your_location_id>
+
+The away mode service is exposed at streamlabswater.set_away_mode where the
+required parameter away_mode should be set to either home or away.
+"""
+
+import logging
+
+import voluptuous as vol
+
+from homeassistant.const import CONF_API_KEY
+from homeassistant.helpers import discovery
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['streamlabswater==1.0.1']
+
+DOMAIN = 'streamlabswater'
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_AWAY_MODE = 'away_mode'
+SERVICE_SET_AWAY_MODE = 'set_away_mode'
+AWAY_MODE_AWAY = 'away'
+AWAY_MODE_HOME = 'home'
```

Comment:
Those two constants should be named in a clear way according their content.

Reply:
Do you have an example of what they should be changed to? The modeled the approach after the nest integration for consistency and this is how it declared those constants.

Reply:
why not just `HOME = 'home'` and `AWAY = 'away'` as values for something called `DEVICE_MODE`?

Reply:
Could there be more modes than `away` or `home`? Why not to make it `is_away` ?

Reply:
The notion of being away seemed like something that could eventually be a general concept so I followed the approach used by climate which calls the service set_away_mode. For consistency and ease of search I used the same constant names used in the nest integration. I don't necessarily disagree with the idea that something different might be more optimal, but until someone designs a common architecture to plug in to I thought being consistent with existing integrations and climate was better.

Reply:
I agree to use similar const names as in climate. But in climate "away" is binary. There's no AWAY_MODE_AWAY, AWAY_MODE_HOME etc. I mean "away" notion is binary in its essence. Someone/Something is either away or not. And  climate implements it accordingly, with `turn_away_mode_on()` and `turn_away_mode_off()` and a bool property `is_away_mode_on`

Reply:
@Adminiuga yeah I agree mine looks a bit more like the nest integration than climate in that regard.